src/HospitalLibrary/Appointments/AppointmentService.cs

```diff
@@ -40,5 +41,10 @@ public AppointmentCancelledDTO CancelAppointment(int appointmentId)
             _unitOfWork.AppointmentRepository.Save();
             return retDto;
         }
+        
+        public async Task<IEnumerable<Appointment>> GetUpcomingForDoctor(Doctor doctor)
```

Comment:
You can remove async and await from here, because the service method is returning Task anyway. You should only await it when you want to access IEnumerable<Appointment>